src/SecretsSyncer/SecretsSyncer.php

```diff
@@ -25,17 +26,36 @@ class SecretsSyncer implements SecretsSyncerInterface {
    */
   protected EntityTypeManagerInterface $entityTypeManager;
 
+  /**
+   * The transliteration manager.
+   *
+   * @var \Drupal\Component\Transliteration\TransliterationInterface
+   */
+  protected TransliterationInterface $transliteration;
+
   /**
    * Constructs a new SecretsSyncer object.
    *
    * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
    *   The entity type manager.
+   * @param \Drupal\Component\Transliteration\TransliterationInterface $transliteration
+   *   The transliteration manager.
    */
-  public function __construct(EntityTypeManagerInterface $entityTypeManager) {
+  public function __construct(EntityTypeManagerInterface $entityTypeManager, TransliterationInterface $transliteration) {
     $this->entityTypeManager = $entityTypeManager;
+    $this->transliteration = $transliteration;
     $this->secretsClient = CustomerSecrets::create()->getClient();
   }
 
+  /**
+   * Get machine name for a given key.
+   */
+  protected function getMachineName(string $secretName): string {
+    $transliterated = $this->transliteration->transliterate($secretName);
+    $transliterated = preg_replace('@[^a-z0-9_.]+@', '_', $transliterated);
```

Comment:
Does `transliterate` also convert to lowercase? If so, this looks good.

Reply:
Good call on the lowercase. It doesn't but I just added `strtolower`

Reply:
Might consider just allowing uppercase in the preg_replace instead, though; this change would be breaking for anyone who created secrets with uppercase keys.

Reply:
Didn't consider that yesterday.

Reply:
This is only when the pantheon_secrets module creates the Key entities. This module doesn't really care about the secret name, it will accept anything there but the entity is a config entity, therefore should have a valid machine name. 

It is still able to save with any machine name but then if you attempt to edit it will fail